matchms/exporting/save_as_json.py

```diff
@@ -42,25 +47,27 @@ def save_as_json(spectrums: List[Spectrum], filename: str):
     fingerprint_export_warning(spectrums)
 
     # Write to json file
+    encoder_class = create_spectrum_json_encoder(export_style)
     with open(filename, "w", encoding="utf-8") as fout:
-        json.dump(spectrums, fout, cls=SpectrumJSONEncoder)
+        json.dump(spectrums, fout, cls=encoder_class)
 
 
-class SpectrumJSONEncoder(json.JSONEncoder):
-    # See https://github.com/PyCQA/pylint/issues/414 for reference
-    def default(self, o):
-        """JSON Encoder which can encode a :py:class:`~matchms.Spectrum.Spectrum` object"""
-        if isinstance(o, Spectrum):
-            spec = o.clone().to_dict()
-            if "fingerprint" in spec.keys():
-                del spec["fingerprint"]
-            return spec
-        return json.JSONEncoder.default(self, o)
+def create_spectrum_json_encoder(export_style):
+    class CustomSpectrumJSONEncoder(json.JSONEncoder):
+        def default(self, o):
+            """JSON Encoder for a matchms.Spectrum.Spectrum object"""
+            if isinstance(o, Spectrum):
+                spec = o.clone().to_dict(export_style)
+                spec.pop("fingerprint", None)
+                return spec
+            return super().default(o)
+    return CustomSpectrumJSONEncoder
 
 
 class ScoresJSONEncoder(json.JSONEncoder):
```

Comment:
Shouldn't this be moved to the Scores class, since it is only used there? Now my first impression was that this was part of the saving a spectrum as JSON, while it is only used when saving the scores as JSON.

Reply:
I think that this is so far only used in the saving spectra as JSON part. Unless it is needed elsewhere I think we don't have to move it into Spectrum.

Reply:
Ah, OK, sorry. I also overlooked this.
The `ScoresJSONEncoder` is indeed used in Scores as well. I am fine either way: we can keep it in the save_as_json part or move this one into Scores. The `create_spectrum_json_encoder` should stay in the save_as_json I believe.

Reply:
I thinks we did this at the time to avoid some circular imports... but that might not be an issue anymore.

Reply:
I think the ScoresJSONEncoder is only used in Scores, or is it also used in save_as_json? But the circular imports could be an issue indeed. 

Yes, the create_spectrum_json_encoder should certainly stay in the save_as_json